src/Uno.Extensions.Core/IDispatcher.cs

```diff
@@ -1,18 +1,32 @@
-﻿namespace Uno.Extensions;
+﻿using System.ComponentModel;
 
+namespace Uno.Extensions;
+
+/// <summary>
+/// An abstraction of the dispatcher that is associated to the UI thread.
+/// </summary>
 public interface IDispatcher
 {
+	/// <summary>
+	///  Gets a value that specifies whether the current execution context is on the UI thread.
+	/// </summary>
+	bool HasThreadAccess { get; }
+
+	/// <summary>
+	/// Adds a task to the queue which will be executed on the thread associated with the dispatcher.
+	/// </summary>
+	/// <remarks>This is the raw version which allows to interact with the native dispatcher the fewest overhead possible.</remarks>
+	/// <param name="action">The task to execute.</param>
+	/// <returns>True indicates that the task was added to the queue; false, otherwise.</returns>
+	[EditorBrowsable(EditorBrowsableState.Advanced)] // Applications should prefer to use the ExecuteAsync which allow to track the execution
+	bool TryEnqueue(Action action);
```

Comment:
This makes IDispatcher super confusing for developers. Can we expose on different interface or just internally?

Reply:
Well I disagree : this is the contract that developers are expecting when they are used to deal with the native dispatcher. 

The issue with the default method is that it forces users to wrap every thing in a `Task` which can be costly for nothing. I would rather move the `ExecuteAsync` to be an extensions method.

BTW this method is flagged as `[EditorBrowsable(EditorBrowsableState.Advanced)]` so it won't be the first suggested.